src/command_modules/azure-cli-interactive/azclishell/telemetry.py

```diff
@@ -4,96 +4,107 @@
+        self.num_scroll_examples = 0
+        self.num_open_config = 0
+        self.num_toggle_default = 0
+        self.num_toggle_symbol_bindings = 0
+        self.num_cli_commands_used = 0
+
+    def get_interactive_session_properties(self):
         properties = {}
-        set_custom_properties(properties, 'starttime', str(self.start_time))
-        set_custom_properties(properties, 'endtime', str(self.end_time))
-
-        subprocess.Popen([
-            sys.executable,
-            os.path.realpath(__file__),
-            json.dumps(properties)])
-
-
-def scrub(text):
-    """ scrubs the parameter values from args """
-    args = parse_quotes(text)
-    next_scrub = False
-    values = []
-    for arg in args:
-        if arg.startswith('-'):
-            next_scrub = True
-            values.append(arg)
-        elif next_scrub:
-            values.append('*****')
-        else:
-            values.append(arg)
-    return ' '.join(values)
-
-
-# TODO: restore this wonky telemetry thing...
-# if __name__ == '__main__':
-#    # If user doesn't agree to upload telemetry, this scripts won't be executed. The caller should control.
-#    SHELL_TELEMETRY.track_event('az/interactive/run', json.loads(sys.argv[1]))
-#    SHELL_TELEMETRY.flush()
+        telemetry_core.set_custom_properties(properties, 'StartTime', str(self.interactive_start_time))
+        telemetry_core.set_custom_properties(properties, 'EndTime', str(self.interactive_end_time))
+        telemetry_core.set_custom_properties(properties, 'OutsideGestures', self.num_outside_gesture)
+        telemetry_core.set_custom_properties(properties, 'ExitGestures', self.num_exit_code_gesture)
+        telemetry_core.set_custom_properties(properties, 'QueryGestures', self.num_query_gesture)
+        telemetry_core.set_custom_properties(properties, 'ScopeChanges', self.num_scope_changes)
+        telemetry_core.set_custom_properties(properties, 'TutorialRuns', self.num_ran_tutorial)
+        telemetry_core.set_custom_properties(properties, 'ExampleScrollingActions', self.num_scroll_examples)
+        telemetry_core.set_custom_properties(properties, 'ConfigurationChanges', self.num_open_config)
+        telemetry_core.set_custom_properties(properties, 'DefaultToggles', self.num_toggle_default)
+        telemetry_core.set_custom_properties(properties, 'SymbolBindingToggles', self.num_toggle_symbol_bindings)
+        telemetry_core.set_custom_properties(properties, 'CliCommandsUsed', self.num_cli_commands_used)
+        return properties
+
+
+_session = Telemetry()
+
+
+# core telemetry operations
+def start():
+    telemetry_core.start()
+
+
+def flush():
+    telemetry_core.flush()
+
+
+def conclude():
+    _session.interactive_end_time = datetime.datetime.now()
+    interactive_session_properties = _session.get_interactive_session_properties()
+    telemetry_core.add_event('interactive', interactive_session_properties)
+    telemetry_core.suppress_new_events()
```

Comment:
If I understand correctly, you send start and end time but not duration?

Reply:
Yes, it seems a little redundant.
Since the regular cli commands also include duration ("TimeSinceSessionStart"), however, I can do the same to be consistent.